clusterman/migration/event.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2019 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import enum
+import json
+from typing import List
+from typing import NamedTuple
+from typing import Union
+
+
+class ConditionTrait(enum.Enum):
+    KERNEL = "kernel"
+    LSBRELEASE = "lsbrelease"
+    INSTANCE_TYPE = "instance_type"
+    UPTIME = "uptime"
+
+
+class MigrationCondition(NamedTuple):
+    trait: ConditionTrait
+    target: Union[str, int, List[str]]
+
+
+class MigrationEvent(NamedTuple):
+    event_id: str
+    event_receipt: str
+    cluster: str
```

Comment:
Why may we need clustername in the event?

Reply:
We currently have a separate queue for all clusters. It would probably be better to keep them as is. By the way CLIs are also different for cluster. I mean we know about cluster. when send/receive event

Reply:
`node migration` instance will also run on specific cluster. IMHO For now it would be better for each cluster to have its own `migration` so that they are all standard.

Reply:
true that we'll have things separated by cluster in all places, but at the same time it doesn't hurt to have this information replicated in the events, it's just extra insurance.